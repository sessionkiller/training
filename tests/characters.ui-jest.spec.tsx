import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import CharacterList from "../pages";

const mockData = {
  results: [
    { id: 1, name: "Rick Sanchez", image: "rick.png" },
    { id: 2, name: "Morty Smith", image: "morty.png" },
  ],
  info: {
    next: "https://rickandmortyapi.com/api/character?page=2",
    prev: null,
  },
};

test("renders character list", () => {
  render(<CharacterList initialData={mockData} />);
  expect(screen.getByText("Rick Sanchez")).toBeInTheDocument();
  expect(screen.getByText("Morty Smith")).toBeInTheDocument();
});

test("handles pagination buttons", async () => {
  global.fetch = jest.fn((): any =>
    Promise.resolve({
      json: () => Promise.resolve(mockData),
    })
  );

  const { getByText } = render(<CharacterList initialData={mockData} />);
  const nextButton = getByText("Next");
  nextButton.click();
  expect(global.fetch).toHaveBeenCalledWith(mockData.info.next);
});

test("matches snapshot", () => {
  const { asFragment } = render(<CharacterList initialData={mockData} />);
  expect(asFragment()).toMatchSnapshot();
});

// test("is accessible", async () => {
//   const { container } = render(<CharacterList initialData={mockData} />);
//   const results = await axe(container);
//   expect(results).toHaveNoViolations();
// });
