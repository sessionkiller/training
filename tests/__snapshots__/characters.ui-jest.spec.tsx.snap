// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Rick and Morty Characters
    </h1>
    <div
      style="display: grid; grid-template-columns: repeat(auto-fill, minmax(150px, 1fr)); gap: 10px;"
    >
      <div>
        <img
          alt="Rick Sanchez"
          src="rick.png"
          style="width: 100%;"
        />
        <p>
          Rick Sanchez
        </p>
      </div>
      <div>
        <img
          alt="Morty Smith"
          src="morty.png"
          style="width: 100%;"
        />
        <p>
          Morty Smith
        </p>
      </div>
    </div>
    <div>
      <button
        disabled=""
      >
        Previous
      </button>
      <button>
        Next
      </button>
    </div>
  </div>
</DocumentFragment>
`;
